src/services/cohortCreation/index.ts

```diff
@@ -16,10 +23,25 @@ const getPatients = memoize(
     if (!query) return []
     const response = await apiFhir.get<FHIR_API_Response<IPatient>>(query)
     const patients = getApiResponseResources(response)
-    return patients ? patients.map((patient) => patient.id) : []
+    return patients?.map((patient) => patient.id) ?? []
   }
 )
 
+const getPatientsFromDocuments = memoize(
+  async (query: string, filter: string): Promise<(string | undefined)[]> => {
+    const response = await apiFhir.get<FHIR_API_Response<IDocumentReference>>(query)
+    const documents = getApiResponseResources(response)
+    const patients = documents
+      ?.map((document) => last(document.subject?.reference?.split('/')))
+      .filter((patient) => !!patient)
+    if (!patients?.length) return []
+
+    const allPatients = await getPatients(`/Patient?${filter}`)
+    return patients?.filter((patient) => allPatients.includes(patient)) ?? []
+  },
+  (a, b) => a + b
```

Comment:
what's this ?

Reply:
a callback to build the cache key https://lodash.com/docs/4.17.15#memoize

Reply:
ok